.github/workflows/build_and_push.yml

```diff
@@ -0,0 +1,95 @@
+        with:
+          # Enables host networking which allows tests run by buildx to access
+          # the containers started by docker compose on localhost
+          driver-opts: network=host
+      # Login to the registry
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.CI_PAT }}
+      # This creates a cache for the current PR, if none exists then
+      # the cache from the most recent PR will be used.
+      - name: Setup Docker layer Cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-docker-${{ github.event.number }}
+          restore-keys: ${{ runner.os }}-docker-
+      # Use curl to fetch a golang or python `Dockerfile` from `mailgun/workflows`
+      # as the github action yaml you are now reading should be exactly the same for
+      # both golang and python builds, only the `Dockerfile` is different.
+      - name: Download the latest buildx Docker file
+        run: |
+          curl -L https://api.github.com/repos/mailgun/workflows/contents/go/Dockerfile \
+          -H "Authorization: token ${{ secrets.CI_PAT }}" \
+          -H "Accept: application/vnd.github.v3.raw" \
+          -o .github/Dockerfile
+      # Start dependent services in containers, NOTE: Some services might not
+      # have external dependencies so this action might need to be removed.
+      - name: Docker Compose
+        run: docker-compose up -d
+      # Automagically extract useful information from the current github context and creates
+      # a set of labels for use by build-push-action to be attached to the final image.
+      - name: Extract Metadata for Docker
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+        id: meta
+      # This action runs Buildx, which allows for a more complex Dockerfile.
+      # We use a buildx Dockerfile to run tests as well as build the final image.
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          # We change the path context here since the github context does not include
+          # changes to local files, like when we download `Dockerfile`.
+          # See https://github.com/docker/build-push-action#git-context
+          context: .
+          file: .github/Dockerfile
+          tags: ghcr.io/${{ github.repository }}:PR${{ github.event.number }}
+          # We use local cache type, so we can clean up the cache
+          # https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#local-cache
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+          ssh: default=/tmp/ssh_agent.sock
+          labels: ${{ steps.meta.outputs.labels }}
+          push: true
```

Comment:
This is a library. What are we going to push?

Reply:
I'm just trying to get the tests to run since they fail when I run them locally.  The defaults aren't suitable, but this repo doesn't seem to be configured for Github Actions yet so nothing works.

Reply:
Tests fail due to the bugs in the code.